proxy/http2/Http2ConnectionState.cc

```diff
@@ -1146,7 +1149,6 @@ Http2ConnectionState::release_stream(Http2Stream *stream)
       ink_assert(client_streams_out_count > 0);
       --client_streams_out_count;
     }
-    stream_list.remove(stream);
```

Comment:
Now `release_stream()` doesn't **release** stream ;) We need some cleanup here like a) move counter updates outside of this function, b) remove argument, and c) rename this function. 
But fixing the crash is urgency, so I'm fine with this changes.